server/models/service-instance.js

```diff
@@ -66,6 +67,32 @@ module.exports = function(ServiceInstance) {
     });
   });
 
+  function recordInstanceInfo(instanceId, instInfo, callback) {
+    ServiceInstance.findById(instanceId, function(err, instance) {
+      if (err) return callback(err);
+      instance.currentDeploymentId = instInfo.commitHash;
+      instance.startTime = new Date();
+      instance.started = true;
+      instance.applicationName = instInfo.appName;
+      instance.npmModules = instInfo.npmModules;
+      instance.PMPort = instInfo.PMPort;
+      instance.containerVersionInfo = {
+        os: {
+          platform: os.platform(),
```

Comment:
This would be handled different on mesh. But for PM, does this really make a difference?